exist-core/src/main/java/org/exist/xquery/XPathException.java

```diff
@@ -153,30 +151,24 @@ public XPathException(XQueryAST ast, ErrorCode errorCode, String message) {
         }
     }
 
-    /**
-     * @param cause the cause throwable
-     * @deprecated Use a constructor with errorCode
-     */
-    @Deprecated
-    public XPathException(final Throwable cause) {
-        super(cause);
-        if(cause instanceof XPathErrorProvider) {
-            this.errorCode = ((XPathErrorProvider)cause).getErrorCode();
-        }
-    }
-
     /**
      * @param cause the cause throwable
      * @param message the error message
      * @deprecated Use a constructor with errorCode
      */
     @Deprecated
-    public XPathException(final String message, final Throwable cause) {
```

Comment:
I expect incompatibilities with XAR files

Reply:
@dizzzz: I re-added this deprecated constructor.